src/pages/App.tsx

```diff
@@ -100,44 +97,30 @@ export default function App(): ReactElement {
                               <Switch>
                                 <Route exact path="/" component={Swap} />
                                 <Route exact path="/pools" component={Pools} />
-                                {pools.map(({ name }) => (
-                                  <Route
-                                    exact
-                                    path={`/pools/${name}/deposit`}
-                                    render={(props) => (
-                                      <Deposit {...props} poolName={name} />
-                                    )}
-                                    key={`${name}-name-deposit`}
-                                  />
-                                ))}
-                                {pools.map(({ name, route }) => (
-                                  <Route
-                                    exact
-                                    path={`/pools/${route}/deposit`}
-                                    render={(props) => (
-                                      <Deposit {...props} poolName={name} />
-                                    )}
-                                    key={`${route}-route-deposit`}
-                                  />
-                                ))}
-                                {pools.map(({ name }) => (
+                                {pools.map((pool) => (
```

Comment:
A better way to do this might just be to have a dynamic route instead of an explicit one. That way instead of creating one route for each pool (and thus having to wait for all pools to be loaded, which might not work), we just have two routes for deposit and withdraw.

```
<Route exact path="/pools/:poolName/deposit" ...>
```

Then use `match.params.poolName` in the child to get pool data.

Reply:
this is a good idea to be explored in a follow-up PR. It might work well with react-query. I will look into it and make a ticket for now.